### PR TITLE
NETOBSERV-121 tab pages

### PR DESCRIPTION
**Pull request by jpinsonneau**
Add details pages for Pod / Service / Namespace / Deployment

Automatically set filters from selected object infos

image

### CODE REVIEW

**Comment by jotak** [comment]
Really nice!
I'd put it on hold in order to gather some feedback on the feature, but it's already super interesting as it is.
/hold

**Comment by jotak** [comment]
/lgtm
we must create a follow-up task to finish the work once https://issues.redhat.com/browse/NETOBSERV-73 is done

**Comment by jotak** [comment]
Follow-up created: https://issues.redhat.com/browse/NETOBSERV-156

**Comment by jpinsonneau** [comment]
Updated from feedbacks : 
- filters cannot be updated in tabs, `chips` show current forced filters
- added 'Edit filter' button to redirect to `/netflow-traffic` page keeping url arguments (filters / refresh)
- in case of 'No result found', clear all filter button will also redirect to `/netflow-traffic` removing all arguments

File: web/src/utils/router.ts [hunk 1]
```diff
@@ -73,3 +75,11 @@ export const removeQueryArguments = (keys: AnyQueryArgs[]) => {
     history.replace(`${url.pathname}?${params.toString()}${url.hash}`);
   }
 };
+
+export const navigate = (pathName: string, keepParams = true) => {
```
Comment (jotak) [inline web/src/utils/router.ts:79]: it seems that this `navigate` triggers a full reload of the console, if possible we should probably avoid that (but i don't know if there are other options)

Comment (jotak) [reply web/src/utils/router.ts:79]: (I think this is what react-router is for)

Comment (jpinsonneau) [reply web/src/utils/router.ts]: Thanks, I used 'react-router-dom' `useHistory` hook :+1:

File: web/src/utils/router.ts [hunk 2]
```diff
@@ -43,13 +44,28 @@ const reporterToFlowdir = _.invert(flowdirToReporter);
...
+  forcedFilters: Filter[] | null = null,
   range: number | TimeRange,
   opts: QueryOptions
 ): QueryArguments => {
   // Note: at the moment the browser query params and API query params are tied together;
   // we may want to decouple them in the future.
   const params: QueryArguments = {};
-  _.each(filters, (f: Filter) => {
+  let allFilters: Filter[];
+  if (forcedFilters) {
```
Comment (jotak) [inline web/src/utils/router.ts:55]: is there cases where we have `forcedFilters` AND some extra `filters` ? I thought if not, we could simplify and, from the caller, pass either `filters` or `forcedFilters`, but not both...

**Comment by jotak** [comment]
looks like filtering by namespace is broken :/ (maybe it's me...)

File: web/src/utils/router.ts (hunk 2 above)
Comment (jpinsonneau) [reply web/src/utils/router.ts:55]: `FlowDirection` and `Limit` are editable filters from the tabs now; so yes you can have both !

However you are not supposed to have forced & normal filters on the same columns but the user may try to edit URL arguments right ?

**Comment by jpinsonneau** [comment]
> looks like filtering by namespace is broken :/ (maybe it's me...)

Hmm good catch ! I also reproduce from main branch.

This seems to be related to `type QueryArguments = { [k in AnyQueryArgs]?: unknown };` that produce wrong params key `SrcNamespace[]` instead of `SrcNamespace`

File: web/src/utils/router.ts (hunk 2 above)
Comment (jotak) [reply web/src/utils/router.ts:55]: But `FlowDirection` and `Limit` are not passed through `filters`, but through `opts`. They should not affect the search filters

Comment (jpinsonneau) [reply web/src/utils/router.ts:55]: You're right, I will simplify that then :+1: 

Thanks for the feedback

**Comment by jotak** [comment]
/lgtm
(rebased locally & tested)

**Review by jotak — APPROVED** [review]

**Pull request merged**